provider/gpkg/gpkg_register.go

```diff
@@ -82,29 +83,74 @@ func AutoConfig(gpkgPath string) (map[string]interface{}, error) {
 	return conf, nil
 }
 
-func extractColsFromSQL(sql string) []string {
-	// Get names of all columns
-	// Extract column definitions from sql.
-	colDefsPattern := `CREATE.*?\(\s+(.*)$`
-	colDefsFinder := regexp.MustCompile(colDefsPattern)
+// extractColsAndPKFromSQL extracts all column names and the primary key colum
+// from an SQL definition string.
+func extractColsAndPKFromSQL(sql string) ([]string, string) {
+	defs := extractColDefsFromSQL(sql)
 
-	colPattern := `"(.+?)"`
-	colFinder := regexp.MustCompile(colPattern)
+	var pkCol string
+	colNames := make([]string, 0, len(defs))
 
-	// Get all column names (drop "CREATE TABLE (" portion)
-	colDefs := colDefsFinder.FindStringSubmatch(sql)[1]
-	colMatches := colFinder.FindAllString(colDefs, -1)
+	// match unquoted (`column_name`) or quoted (`"column name"`) indentifiers
+	colFinder := regexp.MustCompile(`^(([a-zA-Z_][a-zA-Z0-9_]*)|"([^"]+)")\s`)
```

Comment:
Would it make sense to have this regex as global var, build up in an init function, so that it does not have to be reinitialized every time this function is called.